x/concentrated-liquidity/math/tick_test.go

```diff
@@ -54,169 +54,198 @@ func PriceToTickRoundDownSpacing(price osmomath.BigDec, tickSpacing uint64) (int
 			tickIndex:     1000100,
-			expectedPrice: osmomath.MustNewDecFromStr("2.0001"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("2.0001"),
 		},
 		"One thousandth cent increments at the tens place: 1": {
 			tickIndex:     9200100,
-			expectedPrice: osmomath.MustNewDecFromStr("12.001"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("12.001"),
 		},
 		"One thousandth cent increments at the tens place: 2": {
 			tickIndex:     9200200,
-			expectedPrice: osmomath.MustNewDecFromStr("12.002"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("12.002"),
 		},
 		"One cent increments at the hundreds place: 1": {
 			tickIndex:     18320100,
-			expectedPrice: osmomath.MustNewDecFromStr("132.01"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("132.01"),
 		},
 		"One cent increments at the hundreds place: 2": {
 			tickIndex:     18320200,
-			expectedPrice: osmomath.MustNewDecFromStr("132.02"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("132.02"),
 		},
 		"Ten cent increments at the thousands place: 1": {
 			tickIndex:     27732100,
-			expectedPrice: osmomath.MustNewDecFromStr("1732.10"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("1732.10"),
 		},
 		"Ten cent increments at the thousands place: 2": {
 			tickIndex:     27732200,
-			expectedPrice: osmomath.MustNewDecFromStr("1732.20"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("1732.20"),
 		},
 		"Dollar increments at the ten thousands place: 1": {
 			tickIndex:     36073200,
-			expectedPrice: osmomath.MustNewDecFromStr("10732"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("10732"),
 		},
 		"Dollar increments at the ten thousands place: 2": {
 			tickIndex:     36073300,
-			expectedPrice: osmomath.MustNewDecFromStr("10733"),
+			expectedPrice: osmomath.MustNewBigDecFromStr("10733"),
 		},
 		"Max tick and min k": {
 			tickIndex:     342000000,
-			expectedPrice: types.MaxSpotPrice,
+			expectedPrice: types.MaxSpotPriceBigDec,
 		},
-		"Min tick and max k": {
-			tickIndex:     types.MinInitializedTick,
-			expectedPrice: types.MinSpotPrice,
+		"tickIndex is MinInitializedTickV1": {
+			tickIndex: types.MinInitializedTick,
+			// 1 order of magnitude below min spot price of 10^-12 + 6 orders of magnitude smaller
+			// to account for exponent at price one of -6.
+			expectedPrice: types.MinSpotPriceBigDec,
+		},
+		"tickIndex is MinCurrentTickV1": {
+			tickIndex: types.MinCurrentTick,
+			// 1 order of magnitude below min spot price of 10^-12 + 6 orders of magnitude smaller
+			// to account for exponent at price one of -6.
+			expectedPrice: types.MinSpotPriceBigDec.Sub(osmomath.BigDecFromDec(osmomath.SmallestDec()).Quo(bigTenDec)),
+		},
+		"tickIndex is MinInitializedTickV2": {
+			tickIndex:     types.MinInitializedTickV2,
+			expectedPrice: types.MinSpotPriceV2,
+		},
+		"tickIndex is MinCurrentTickV2": {
+			tickIndex:     types.MinCurrentTickV2,
+			expectedPrice: types.MinSpotPriceV2,
+		},
+		"tickIndex is MinInitializedTick + 1 ULP": {
```

Comment:
just out of curiosity, what does ULP here stand for?

Reply:
Unit in the last place https://en.wikipedia.org/wiki/Unit_in_the_last_place